x-team/xp-ui
# Make TwoColumnsLayout sidebar collapsible

**Release Type:** *Non-breaking changes*

Fixes https://x-team-internal.atlassian.net/browse/XP-3571

## Description

Make TwoColumnsLayout sidebar collapsible.

## Checklist


## Steps to Test or Reproduce

Try this changes in XP (copy XP-UI lib files over XP node_modules).

1. Open http://localhost:8000/admin/search on mobile viewport and observe it loads the page with sidebar collapsed
1. Toggle the sidebar by clicking in the splitter
1. Also try out https://deploy-preview-503--xpui-develop.netlify.com/?path=/story/screens-and-layouts-twocolumnslayout-use-cases--composed-in-adminscreen-with-filters-and-sidebarheading-element

To see all details of the implementation, please take a look at this demo https://youtu.be/ZNYKln-_GTo, also make sure to read my comment in the ticket https://x-team-internal.atlassian.net/browse/XP-3571?focusedCommentId=22560&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-22560 explaining why AC 2 and 5 are not covered. (at this moment I proposed to remove these items, still waiting feedback)

## Impacted Areas in Application

`TwoColumnsLayout`

## Screenshots

![2020-03-06 01 59 36](https://user-images.githubusercontent.com/131859/76054390-40a79100-5f4f-11ea-921b-396916fb83ba.gif)
